### PR TITLE
Install script

### PR DESCRIPTION
**Pull request by dominikl**
Based on  #8

Combines the various build/install steps in an `install.R` script (which could be used for an `romero.gateway` docker image?)
Given that you installed R, rJava and Maven, you should be able to automatically build and install the `romero.gateway` R package by running `R CMD BATCH install.R` (for testing this PR, set git repo URL/branch in install.R to this branch first!).

### CODE REVIEW

**Comment by bramalingam** [comment]
Tested locally and getting the following error on install,

Installing packages into ‘/Users/bramalingam/Library/R/3.3/library’
(as ‘lib’ is unspecified)
Error in contrib.url(repos, type) : 
  trying to use CRAN without setting a mirror
Calls: install.packages -> grep -> contrib.url
Execution halted

**Comment by dominikl** [comment]
Issue should be fixed now. Also note, for testing **the git repo URL in `install.R` has to be set to this branch first**, otherwise it'll use the master branch and fail.

File: README.md [hunk 1]
```diff
@@ -1,34 +1,60 @@
...
+
 * Download this repository: 
   * Using Git: ```git clone https://github.com/ome/rOMERO-gateway.git```
   * _Alternative_: Download as Zip and extract.
-* ```cd``` into the ```rOMERO``` directory
+* ```cd``` into the ```rOMERO-gateway``` directory
 * Download the dependencies
   * Using Maven: Run ```mvn install```
-  * _Alternative_: Create ```lib``` directory. Download [OMERO.Insight client](http://downloads.openmicroscopy.org/omero/5.3.0/). Extract the zip file. Copy all files within ```libs``` directory into the previously created ```rOMERO/lib``` directory
+  * _Alternative_: Create ```inst/java``` directory. Download [OMERO.Insight client](http://downloads.openmicroscopy.org/omero/5.3.0/). Extract the zip file. Copy all files within ```libs``` directory into the previously created ```rOMERO/inst/java``` directory
```
Comment (jburel) [inline README.md]: OMERO.insight

**Comment by dominikl** [comment]
Would that be ok to merge @bramalingam , @jburel ? 
I'd have another PR ready which adds lots of additional methods and tests e.g. for dealing plates/well, but which on builds on top of this PR.

**Comment by jburel** [comment]
@bramalingam did you test the last changes?

**Comment by bramalingam** [comment]
`Installing package into ‘/Users/bramalingam/Library/R/3.3/library’
(as ‘lib’ is unspecified)
* installing *source* package ‘romero.gateway’ ...
** R
** inst
** preparing package for lazy loading
Error : .onLoad failed in loadNamespace() for 'rJava', details:
  call: dyn.load(file, DLLpath = DLLpath, ...)
  error: unable to load shared object '/Users/bramalingam/Library/R/3.3/library/rJava/libs/rJava.so':
  dlopen(/Users/bramalingam/Library/R/3.3/library/rJava/libs/rJava.so, 6): Library not loaded: @rpath/libjvm.dylib
  Referenced from: /Users/bramalingam/Library/R/3.3/library/rJava/libs/rJava.so
  Reason: image not found
Error : package ‘rJava’ could not be loaded
ERROR: lazy loading failed for package ‘romero.gateway’
* removing ‘/Users/bramalingam/Library/R/3.3/library/romero.gateway’
Warning messages:
1: In dir.create(repoDir) :
  '/var/folders/y5/fwnjk4m927bbrrr13sx3spch000cgd/T//RtmpLpq5r9/romero-gateway' already exists
2: In install.packages(packageFile, repos = NULL, type = "source") :
  installation of package ‘/private/var/folders/y5/fwnjk4m927bbrrr13sx3spch000cgd/T/RtmpLpq5r9/romero.gateway_5.3.0.tar.gz’ had non-zero exit status`

**Comment by bramalingam** [comment]
the mvn part seems to be working well, but the rJava installation seems to have issues in the install.R script.

**Comment by dominikl** [comment]
I think you just have to install rJava beforehand, can't install it via script, because it's such a pain. By the way, never managed to install it on OSX anyway.

**Comment by dominikl** [comment]
I just tested it on a fresh linux. Only need to install Maven and R manually via linux package manager. Then run  as root: ```export JAVA_HOME=[path to JVM]; R CMD javareconf``` Then `install.R` script will do the rest (including rJava installation).

**Comment by dominikl** [comment]
@bramalingam 
I found a way to install rJava on OSX:
```
# Update brew
brew update

# Reinstall newest R 
brew reinstall -s r

# Get a root shell
su
bash
export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_91.jdk/Contents/Home/jre # whatever your JDK/JRE path is
R CMD javareconf

# exit bash and root shell
exit
exit

# Install rJava
R
install.packages('rJava')
```

Then update the git repo URL and branch in `install.R` to **point to this branch**.

Then run the script
```
# First make sure maven is in your PATH
export PATH=/Users/dlindner/lib/apache-maven-3.2.3/bin:$PATH # what ever you maven path is

# ... and JAVA_HOME is set
export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_91.jdk/Contents/Home/jre # whatever your JDK/JRE path is

# Run script
R CMD BATCH install.R
```

**Comment by bramalingam** [comment]
Tested on linux and everything works as expected.
My mac had outdated Xcode and command line tools, so couldn't do the testing on the OSX machine.

But tested on Centos7 and was successful.
Looks good to merge.

**Pull request merged**